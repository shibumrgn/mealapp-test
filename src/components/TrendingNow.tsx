import { ScrollView, StyleSheet, Text, View } from 'react-native'
import React, { useEffect, useState } from 'react'
import { IRecipe } from '../interfaces/IRecipe'
import axios from 'axios';
import VideoRecipe from './recipes/VideoRecipe';
import Section from './Section';

const TrendingNow = () => {

    const [recipes, setRecipes] = useState<IRecipe[]>([]);

    const _getTrendingRecipes = async () => {
        const response = await axios.get("https://www.themealdb.com/api/json/v1/1/search.php?f=a");
        if(response?.data?.meals) {
            setRecipes(response.data?.meals);
        }
    }

    useEffect(() => {
      _getTrendingRecipes();
    }, []);
    

    return (
        <Section
            title='Trending Now ðŸ”¥'
            marginLeft={20}
        >
            {recipes?.map(recipe => <VideoRecipe key={recipe?.idMeal} recipe={recipe}/>)}
        </Section>
  )
};

export default TrendingNow;

const styles = StyleSheet.create({});