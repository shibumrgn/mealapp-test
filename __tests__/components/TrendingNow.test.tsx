
import 'react-native';
import React from 'react'; 
import renderer from 'react-test-renderer';
import TrendingNow from '../../src/components/TrendingNow';
jest.useFakeTimers();
jest.mock("axios");
 
it('renders correctly', () => {
    const render = renderer.create(<TrendingNow/>)?.toJSON();
    expect(render).toMatchSnapshot();
});
 