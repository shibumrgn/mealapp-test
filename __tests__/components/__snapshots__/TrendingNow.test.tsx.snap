// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 30,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginBottom": 20,
        "marginLeft": 20,
      }
    }
  >
    Trending Now ðŸ”¥
  </Text>
  <RCTScrollView
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingLeft": 20,
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
